File: Downloads/a2a_coder/scripts/list_free_models.py

# scripts/list_free_models.py
import asyncio
import json
import os
from blackbox_api_mock import list_models

async def main():
    all_models = await list_models()
    free_models = [m for m in all_models if m.get("free") == True]

    print("Found free models:", [m["name"] for m in free_models])

    agent_capabilities = []
    for m in free_models:
        agent_capabilities.append({
            "id": m["name"],
            "max_tokens": m["context_length"],
            "parameters": m["parameters"]
        })

    with open("agent_capabilities.json", "w") as f:
        json.dump(agent_capabilities, f, indent=2)

if __name__ == "__main__":
    asyncio.run(main())

File: Downloads/a2a_coder/scripts/list_free_models.js

// scripts/list_free_models.js
import { listModels } from "./blackbox-ai.js";
import fs from "fs";

async function main() {
  const all = await listModels(); 
  // assume each model has { name, context_length, parameters, free }
  const freeModels = all.filter(m => m.free === true);

  console.log("Found free models:", freeModels.map(m => m.name));
  fs.writeFileSync(
    "agent_capabilities.json",
    JSON.stringify(
      freeModels.map(m => ({
        id: m.name,
        max_tokens: m.context_length,
        parameters: m.parameters
      })),
      null,
      2
    )
  );
}

main().catch(err => {
  console.error(err);
  process.exit(1);
});
